src/OpenLoco/src/Windows/MapWindow.cpp

```diff
@@ -1433,11 +1415,96 @@ namespace OpenLoco::Ui::Windows::MapWindow
         events.drawScroll = drawScroll;
     }
 
-    static void sub_46CFF0()
+    // 0x0046D0C3
+    static uint32_t checkIndustryColours(PaletteIndex_t colour, uint32_t colourMask)
     {
-        registers regs;
-        call(0x0046CFF0, regs);
+        for (auto i = 0; i < 31; i++)
+        {
+            int8_t industryColour = industryColours[i];
+            int8_t diff = std::abs(industryColour - colour);
+
+            if (diff <= 2)
+            {
+                colourMask &= ~(1U << i);
+            }
```

Comment:
This implementation leads to the wrong result, blocking out too many colours. No idea why yet, but I've been staring at this for too long. If I leave it to call vanilla as below, industry colours are assigned properly.
```
registers regs;
regs.al = colour;
regs.ebp = colourMask;

call(0x0046D0C3, regs);
return regs.ebp;
```